File: engine/model/src/main/scala/tools/packet/wedding/WeddingEnd.scala
package tools.packet.wedding

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class WeddingEnd(private var _blessEffect: Boolean, private var _groomId: Int, private var _brideId: Int,
                 private var _step: Byte) extends PacketInput {
  def blessEffect: Boolean = _blessEffect

  def groomId: Int = _groomId

  def brideId: Int = _brideId

  def step: Byte = _step

  override def opcode(): SendOpcode = SendOpcode.WEDDING_CEREMONY_END
}

File: engine/engine-model/src/main/java/tools/packet/wedding/WeddingEnd.java
package tools.packet.wedding;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record WeddingEnd(Boolean blessEffect, Integer groomId, Integer brideId, Byte step) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.WEDDING_CEREMONY_END;
   }
}